src/flint/fmpq_mpoly.jl

```diff
@@ -344,44 +332,32 @@ end
 ###############################################################################
 
 for (jT, cN, cT) in ((QQFieldElem, :fmpq, Ref{QQFieldElem}), (ZZRingElem, :fmpz, Ref{ZZRingElem}),
-                     (Int, :si, Int))
+                     (Int, :si, Int), (UInt, :ui, UInt))
   @eval begin
     function +(a::QQMPolyRingElem, b::($jT))
       z = parent(a)()
-      ccall(($(string(:fmpq_mpoly_add_, cN)), libflint), Nothing,
-            (Ref{QQMPolyRingElem}, Ref{QQMPolyRingElem}, ($cT), Ref{QQMPolyRing}),
-            z, a, b, parent(a))
-      return z
+      return add!(z, a, b)
     end
```

Comment:
This function now only depends on `jt` from the for loop, and this is only in the signature.
I think it would increase readability, if we move it outside of this `@eval` block and give it a signature of `function +(a::QQMPolyRingElem, b::Union{QQFieldElem, ZZRingElem, Int, UInt})`.
Similar things can be applied to many more now coinciding functions in this `for @eval` loop and the new one introduced below.

Reply:
Yes I am aware of that but I deliberately did not yet touch it here, because there is actually a bigger refactoring that will be possible soon. I am busy at the Probebegehung now, so I did not yet have time to write an issue / PR for that.

Background: we also want `add!(a,b,c::Integer)` methods which dispatch to the Int/UInt/ZZRingElem methods depending on the specific type of `c`, and we want them w/o having to write the same boiler plate code again and again (or at least keep the repetition minimal). I have some suggestions for that, but let's discuss it on an issue later (or first on slack, but anyway not now ;-) )